### PR TITLE
Doc for subgraphs.  Example friends graph.

### PR DESCRIPTION
**Pull request by jkbradley**
Doc for subgraphs in user-guide.
Example friends graph available in Scala and Python.

Minor fixes:
* quick-start pyspark shell
* user-guide: graph creation
* Remove println from debugging
* Persist vertices when creating graph from edges only

### CODE REVIEW

**Comment by jkbradley** [comment]
CC @thunterdb I'll follow up to correct some of the to-do items, including methods for getting vertex degrees from Python.

File: docs/user-guide.md [hunk 1]
```diff
@@ -150,21 +225,87 @@ TODO
...
+<div class="codetabs">
+
+<div data-lang="scala"  markdown="1">
+{% highlight scala %}
+import org.graphframes.examples
+val g: GraphFrame = examples.Graphs.friends  // get example graph
+
+// Select subgraph based on edges of type "follow"
+// pointing from a younger user to an older user.
+val paths = g.find("(a)-[e]->(b)")
```
Comment (mengxr) [inline docs/user-guide.md]: `(a)-[e]` -> `(a)->[e]`

File: docs/user-guide.md [hunk 2]
```diff
@@ -150,21 +225,87 @@ TODO
...
+  .filter("e.relationship = 'follow'")
+  .filter("a.age < b.age")
+// "paths"" contains vertex info. Extract the edges.
```
Comment (mengxr) [inline docs/user-guide.md]: nit: `""` -> `"`

File: docs/user-guide.md [hunk 3]
```diff
@@ -150,21 +225,87 @@ TODO
...
+</div>
+
+<div data-lang="python"  markdown="1">
+{% highlight python %}
+from graphframes.examples import Graphs
+g = Graphs(sqlContext).friends()  # Get example graph
+
+# Select subgraph based on edges of type "follow"
+# pointing from a younger user to an older user.
+paths = g.find("(a)-[e]->(b)")\
```
Comment (mengxr) [inline docs/user-guide.md]: `(a)->[e]`

**Comment by mengxr** [comment]
Made one pass, only minor comments. We might consider using the "include_example" tag from Apache Spark for example code (not in this PR).

File: docs/user-guide.md (hunk 1 above)
Comment (jkbradley) [inline docs/user-guide.md]: e is an edge; it's the right syntax

**Comment by jkbradley** [comment]
I thought about using include example, but I figured this was fastest for the short term.

**Comment by jkbradley** [comment]
I'll go ahead and merge this.  Thanks for checking it!

File: docs/user-guide.md (hunk 1 above)
Comment (mengxr) [inline docs/user-guide.md]: Ah, I thought `[..]` already indicates `e` is an edge.

**Pull request merged**